docs/computing/running/toil-cwl-runner.md

```diff
@@ -0,0 +1,115 @@
+- Can also run on other batch systems: Grid Engine, Torque, LSF, HTCondor.
+- Launches and monitors Slurm jobs for you. Also constructs the apptainer commands.
+
+## Disadvantages for using CWL
+
+## Disadvantages for using `toil-cwl-runner`
+- Just a workflow runner. Won't manage your data, or keep track of previous workflow runs.
+
+## Installing `toil-cwl-runner`
+
+Install `toil` with `CWL` plugin.
+```
+cd /projappl/<project_nnnnnnn>
+python -m venv venv
+source venv/bin/activate
+
+pip install -U setuptools wheel
+pip install toil[cwl]
+
+toil-cwl-runner --version
+```
+
+Install `nodejs` which provides helpful tools for debugging `toil` internals.
+```
+cd /projappl/project_nnnnnnn
+wget https://nodejs.org/dist/v18.16.1/node-v18.16.1-linux-x64.tar.xz
+tar -xf node-v18.16.1-linux-x64.tar.xz
+export PATH=$PATH:/projappl/project_nnnnnnn/node-v18.16.1-linux-x64/bin
+```
+
+## Defining CWL workflows
+
+Learning resources
+- [Novice CWL tutorial](https://carpentries-incubator.github.io/cwl-novice-tutorial/), includes detailed setup instructions for local editing and running on Microsoft Windows, macOS, and Linux
+- <https://www.commonwl.org/user_guide/>
+
+## Running CWL workflows with `toil-cwl-runner`
+
+!!! Note
+    Singularity containers can't be run in the **login node** or in an **interactive session** due to network constraints.
+
+When you have defined a workflow with `CWL`, you can send it to the cluster using `sbatch`, and then `toil` will start new jobs for each item in the workflow description.
+
+### Preliminary Steps
+Create working directories for `toil`.
+```
+mkdir /projappl/project_nnnnnnn/<job store name>
+mkdir /projappl/project_nnnnnnn/<work dir name>
+mkdir /projappl/project_nnnnnnn/<tmp name>
+```
+
+### Creating the sbatch file
+The `sbatch` file `workflow.sh` will reference the `CWL` file `workflow.cwl` where you have described your workflow steps.
+
+!!! Note
+    See [batch documentation](./creating-job-scripts-puhti.md) on how to fill out the `#SBATCH` values.
+
+`workflow.sh`
+```bash
+#!/bin/sh
+#SBATCH --job-name=<job name here>
+#SBATCH --account=<project_number here>
+#SBATCH --time=01:00:00
+#SBATCH --mem-per-cpu=1G
+#SBATCH --nodes=1
+#SBATCH --cpus-per-task=2
+#SBATCH --partition=small
+
+source /projappl/<project_nnnnnnn>/venv/bin/activate
+
+WORKDIR=/projappl/project_nnnnnnn
+SCRATCH=/scratch/project_nnnnnnn
+export TMPDIR=$WORKDIR/tmp
+export TOIL_SLURM_ARGS="--account=project_nnnnnnn --partition=small"
+export CWL_SINGULARITY_CACHE=$WORKDIR/singularity
+unset XDG_RUNTIME_DIR
+
+TOIL_SLURM_ARGS="--account=<project_nnnnnnn> --partition=small" toil-cwl-runner \
+    --jobStore $WORKDIR/<job store name> \
+    --workDir $SCRATCH/<work dir name> \
+    --tmpdir-prefix $TMPDIR/<tmp name> \
+    --batchSystem slurm \
+    $WORKDIR/workflow.cwl \
+    --message "message for job"
+```
+
+Send your workflow to the cluster.
+```
+sbatch workflow.sh
+```
+
+## Monitoring a running workflow
```

Comment:
Work in progress? We are using [The WIP GitHub App](https://github.com/wip/app).

Reply:
Sure; I've marked this as a `draft` PR in the GitHub interface.

Reply:
Okay, I think this is ready for reviewing and merging now.